### PR TITLE
Add Image Parallax Scroll View Controller

### PR DESCRIPTION
**Pull request by whoiswillma**
Issue: #216

Add a view controller for parallax scrolling image views.

(There is some artifacting in the GIF below, the visual issues you see at the bottom near the home pill slide thing are not there)

out

### CODE REVIEW

File: Eatery/Controllers/ImageParallaxScrollViewController.swift [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    func loadImage(from url: URL) {
+        imageView.kf.setImage(with: url)
+    }
+
+}
+
+extension ImageParallaxScrollViewController: UIScrollViewDelegate {
+
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        if scrollView.contentOffset.y < 0 {
```
Comment (chalo2000) [inline Eatery/Controllers/ImageParallaxScrollViewController.swift:62]: Just wanted to leave a little reminder that we may get some odd jumpy looking scroll animations if we change the 0 to anything else

**Review by chalo2000 — APPROVED** [review]
definitely much more readable in a separate file, I think this big refactor will be extremely beneficial

File: Eatery/Controllers/ImageParallaxScrollViewController.swift [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import Kingfisher
+import SnapKit
+import UIKit
+
+/**
+ Manage a scroll view and an image view with a parallax effect.
+ */
+class ImageParallaxScrollViewController: UIViewController {
+
+    private let scrollView = UIScrollView(frame: .zero)
```
Comment (kevinchan159) [inline Eatery/Controllers/ImageParallaxScrollViewController.swift]: can just do `UIScrollView()`

File: Eatery/Controllers/ImageParallaxScrollViewController.swift [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    private let imageView = UIImageView(image: nil)
+    
+    let contentView = UIView(frame: .zero)
```
Comment (kevinchan159) [inline Eatery/Controllers/ImageParallaxScrollViewController.swift]: same

File: Eatery/Controllers/ImageParallaxScrollViewController.swift [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+import SnapKit
+import UIKit
+
+/**
+ Manage a scroll view and an image view with a parallax effect.
+ */
+class ImageParallaxScrollViewController: UIViewController {
+
+    private let scrollView = UIScrollView(frame: .zero)
+    private let imageView = UIImageView(image: nil)
```
Comment (kevinchan159) [inline Eatery/Controllers/ImageParallaxScrollViewController.swift]: same

**Review by kevinchan159 — APPROVED** [review]
w o w fancy 🤩

**Pull request merged**